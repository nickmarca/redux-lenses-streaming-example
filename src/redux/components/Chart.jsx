import React, { useState } from 'react';
import { LineChart, Line, CartesianGrid, XAxis, YAxis } from 'recharts';

function Chart({ data }) {
    const [lineKey, setLineKey] = useState('');
    const [xAxisKey, setXAxis] = useState('');

    if (!data.length) {
        return <div />;
    }

    return (
        <div className="panel">
            <div className="panel-block">
                <div
                    style={{
                        marginRight: 15,
                        display: 'flex',
                        alignItems: 'center',
                    }}
                >
                    <label
                        className="label"
                        style={{ marginBottom: 0, marginRight: 7 }}
                    >
                        Select 'Y' Axis
                    </label>
                    <div className="select">
                        <select
                            onChange={({ target }) => setLineKey(target.value)}
                        >
                            {Object.keys(data[0]).map(key => (
                                <option value={key} key={key}>
                                    {key}
                                </option>
                            ))}
                        </select>
                    </div>
                </div>

                <div
                    style={{
                        marginRight: 15,
                        display: 'flex',
                        alignItems: 'center',
                    }}
                >
                    <label
                        className="label"
                        style={{ marginBottom: 0, marginRight: 7 }}
                    >
                        Select 'X' Axis
                    </label>
                    <div className="select">
                        <select
                            onChange={({ target }) => setXAxis(target.value)}
                        >
                            {Object.keys(data[0]).map(key => (
                                <option value={key} key={key}>
                                    {key}
                                </option>
                            ))}
                        </select>
                    </div>
                </div>
            </div>

            {lineKey && xAxisKey ? (
                <LineChart width={600} height={300} data={data}>
                    <Line type="monotone" dataKey={lineKey} stroke="#8884d8" />
                    <CartesianGrid stroke="#ccc" />
                    <XAxis dataKey={xAxisKey} />
                    <YAxis />
                </LineChart>
            ) : null}
        </div>
    );
}

export default Chart;
