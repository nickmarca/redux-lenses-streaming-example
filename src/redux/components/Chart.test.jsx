import React from 'react';
import { create } from 'react-test-renderer';
import Chart from './Chart';

describe('Chart', () => {
    it('matches the snapshot', () => {
        const actions = create(<Chart data={[]} />);

        expect(actions.toJSON()).toMatchSnapshot();
    });
});
