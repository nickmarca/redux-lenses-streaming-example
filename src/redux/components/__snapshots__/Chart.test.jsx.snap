// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart matches the snapshot 1`] = `<div />`;
